components/Searchbar.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+import TripItem from "./TripItem";
+
+export default function Searchbar({ currentTrips }) {
+  const [searchInput, setSearchInput] = useState("");
+
+  const filteredTrips = currentTrips.filter((trip) => {
+    if (searchInput == "") {
+      return true;
+    } else {
+      const searchTerm = searchInput.toLowerCase();
+      return (
+        trip.tripName.toLowerCase().includes(searchTerm) ||
+        trip.tripDestination.toLowerCase().includes(searchTerm)
+      );
+    }
+  });
```

Comment:
I like how you aren't using a second state here. This is much cleaner than creating another state and setting it via useEffect (which is very dirty). Well done!

By the way, this is called _derived state._ The `filteredTrips` array is derived from a state (in this case `currentTrips`) and changes every time `searchInput` changes through user interaction with the input field.